server/bin/pbench-dispatch.py

```diff
@@ -0,0 +1,405 @@
+                tar = tarfile.open(name=os.path.realpath(result), mode="r")
+                tar.extractall(path=income_unpack)
+                tar.close()
+        else:
+            print("$TS: 'tar -xf $result' failed: code $status | tee -a $mail_content >&4")
+            nerrs=nerrs+1
+
+        dest_path = os.path.join(archive,hostname,resultname,"")
+
+        src = income_unpack
+        if os.path.exists(src):            
+            shutil.move(src, dest_path)
+        else:
+            nerrs = nerrs+1
+            os.remove(income_unpack)
+
+        try:
+            if os.path.islink(os.path.join(archive,hostname,linkdest,resultname)):
+                print("success")
+        except Exception:
+            print("$TS: Cannot move symlink to $ARCHIVE/$hostname/$resultname.tar.xz from $linksrc to $linkdest: code $status | tee -a $mail_content >&4")
+            nerrs=nerrs+1
+            os.remove(income)
+            continue
+
+        resultname = resultname+".tar.xz"
+        src_path = os.path.join(archive,hostname,linksrc,resultname)
+        if os.path.exists(src_path):
+            dest_path = os.path.join(archive,hostname,linkdest, resultname)
+            os.makedirs(dest_path)
+            if os.path.exists(dest_path):
+                shutil.copy(src_path, dest_path)
+
+                if not os.path.exists(dest_path):
+                    print("Cannot move symlink from linksrc to linkdest")
+                    nerrs=nerrs+1
+
+        state_list = []
+        controller_path = os.path.join(archive,hostname)
+        controller_link = os.path.join(archive,hostname,resultname)
+        state_list = copy_symlinks(state_list, to_un_linkedestlist, controller_path, controller_link)
+        
+
+        totsuc, toterr = state_list[0], state_list[1]
+
+        if (toterr > 0 ):
+            # Count N link creations as one error since it is for handling of a
+            # single tarball.
+            nerrs = nerrs+1
+
+        end_time=time()
+        duration=end_time-start_time
+        
+        # log the success
+        print("$TS: $hostname/$resultname: success - elapsed time (secs): $duration - size (bytes): $size")
+        ntb=ntb+1
+
+    return Results(ntotal=ntotal,
+                   ntb=ntb,
+                   nerrs=nerrs,
+                   ndups=ndups,
+                   ninfo=ninfo)
+
+
+def main():
+
+    cfg_name = os.environ.get("CONFIG")
+    if not cfg_name:
+        print("{}: ERROR: No config file specified; set CONFIG env variable or"
+                " use --config <file> on the command line".format(_NAME_),
+                file=sys.stderr)
+        return 2
+
+    try:
+        config = PbenchConfig(cfg_name)
+    except BadConfig as e:
+        print("{}: {}".format(_NAME_, e), file=sys.stderr)
+        return 1
+
+    logger = get_pbench_logger(_NAME_, config)
+
+    qdir = config.QDIR = config.get('pbench-server', 'pbench-quarantine-dir')
+
+    lb_obj = LocalConfigObject(config)
```

Comment:
In lines 344-349  you don't need lb_obj. As mentioned above, you can get rid of it in the sanity_check() function and after that, it is not used at all.